### DISCUSSION TITLE
Compilation failure using mut on a navigation controller

### DISCUSSION THREAD

**Discussion by mycroftcanner** [pointfreeco/swift-overture#65] [Q&A]
A bit confused with the following code:

```swift
  let baseNavigationBarStyle: (UINavigationBar) -> Void = { ... }

    var navigationController: UINavigationController = update(
      UINavigationController(rootViewController: controller),
      mver(\.navigationBar, baseNavigationBar)
    )
```

fails to compile:

> Key path value type 'WritableKeyPath<UINavigationController, UINavigationBar>' cannot be converted to contextual type 'KeyPath<UINavigationController, UINavigationBar>'

The following works fine:

```swift

let baseNavigationBarStyle: (UINavigationBar) -> Void = { ... }

let baseNavigationBar: (inout UINavigationBar) -> Void = {
  with($0, baseNavigationBarStyle)
}
var navigationController = UINavigationController(rootViewController: controller)

with(navigationController.navigationBar, baseNavigationBarStyle)
```

**Comment by stephencelis** [comment]
Sorry for the delay in any response. Since this doesn't seem to be a bug and more of a UIKit quirk, I'm going to move this to a discussion. The problem would seem to lie in the fact that `\.navigationBar` is a read-only property, while any writable key path chained along it will become a reference-writable key path.

There may be some key path compositional lessons to learn here, like being able to `mver(\.value) where Value: AnyObject`, but because that will increase the overload space, let's keep things as a discussion for now.
